### PR TITLE
Add support for DNG and other RAW formats

### PR DESCRIPTION
**Pull request by narogers**
This patch enables support in hydra-derivatives for image formats that ImageMagick can read but not write. Examples would be RAW files such as those generated by cameras although a full list is available at http://www.imagemagick.org/script/formats.php. If the order of operations is reversed (format then scale or resize) it might be possible to completely do away with the :raw_image processor and just merge it into :image although benchmarking locally does show some performance penalties with the first approach.

### CODE REVIEW

**Comment by narogers** [comment]
Adding test cases for RAW images in the next day or so as soon as I can create some smaller test images

File: lib/hydra/derivatives/image.rb [hunk 1]
```diff
@@ -33,7 +34,7 @@ def new_mime_type(format)
 
       def create_resized_image(output_file, size, format, quality=nil)
         create_image(output_file, format, quality) do |xfrm|
-          xfrm.resize(size) if size.present?
+          xfrm.scale(size) if size.present?
```
Comment (mjgiarlo) [inline lib/hydra/derivatives/image.rb]: Can you say a bit about this change?

File: lib/hydra/derivatives/image.rb [hunk 2]
```diff
@@ -53,10 +54,19 @@ def write_image(output_file, xfrm)
         output_file.content = stream
       end
 
-      # Override this method if you want a different transformer, or need to load the
-      # raw image from a different source (e.g.  external file)
+      # Override this method if you want a different transformer, or # need to load the raw image from a different source (e.g.
```
Comment (mjgiarlo) [inline lib/hydra/derivatives/image.rb]: Should these changes be in the `Image` class, or in an overridden `#load_image_transformer` method in the `RawImage` class?

**Comment by mjgiarlo** [comment]
Can you also add test coverage in `spec/units/raw_image_spec.rb`? (You can crib from image_spec.rb to get started.)

File: lib/hydra/derivatives/image.rb (hunk 1 above)
Comment (narogers) [inline lib/hydra/derivatives/image.rb]: Benchmarking scale versus resize it turns out the former is a bit faster (see http://www.imagemagick.org/Usage/resize/ for more technical details). For thumbnail sized images the difference in quality may not matter. I could also be just as convinced to change it back to resize since it was just a slight performance tweak.

File: lib/hydra/derivatives/image.rb (hunk 2 above)
Comment (narogers) [inline lib/hydra/derivatives/image.rb]: I can go either way although it does not hurt the image class to provide the hinting for other formats.

Comment (mjgiarlo) [inline lib/hydra/derivatives/image.rb]: I'd say revert this change unless it substantially helps you deliver RAW support in this PR.

File: lib/hydra/derivatives/image.rb (hunk 1 above)
Comment (mjgiarlo) [inline lib/hydra/derivatives/image.rb]: I'd say revert this change unless it substantially helps you deliver RAW support in this PR.

Comment (narogers) [inline lib/hydra/derivatives/image.rb]: Done and will be pushed once I can get functioning tests

File: lib/hydra/derivatives/image.rb (hunk 2 above)
Comment (narogers) [inline lib/hydra/derivatives/image.rb]: I will pull that out into raw_image

**Comment by narogers** [comment]
While I am in the spec file I noticed lots of let ... tap entries. Should I just clean these up as part of this PR or create a new one next week? https://github.com/narogers/hydra-derivatives/blob/master/spec/units/transcoding_spec.rb

**Comment by narogers** [comment]
Not sure how this got closed in the process of doing housecleaning. I will reopen it with a reference back to here.
